apitests/spec/devices_spec.ts

```diff
@@ -30,8 +30,102 @@ var findDeviceWithCommandHistory = function(devices:Devices) {
+    expect(device._ts).toBeTruthy();
+    expect(device._attachments).toBeTruthy();
+}
+
+var checkRequiredPropertiesEnabledDevice = function(device:DeviceInfo) {
+    expect(device).toBeTruthy();
+    expect(device.DeviceProperties.HubEnabledState).not.toBeNull();
+    expect(device.Telemetry).toBeTruthy();
+    expect(device.Version).toBeTruthy();
+    expect(device.ObjectType).toBeTruthy();
+    expect(device.IoTHub).toBeTruthy();
+}
+
+var checkNoCommandsDisabledDevice = function(device:DeviceInfo) {
+    expect(device).toBeTruthy();
+    expect(device.Commands).toBeTruthy();
+    expect(device.Commands.length).toEqual(0);
+    expect(device.DeviceProperties.HubEnabledState).toBeNull();            
+}
+
+var checkCommandsEnabledDevice = function(device:DeviceInfo) {
+    expect(device).toBeTruthy();
+    expect(device.DeviceProperties.HubEnabledState).not.toBeNull();
+    expect(device.Commands).toBeTruthy();
+    expect(device.Commands.length).toBeGreaterThan(0);
+    expect(device.Commands[0]).toBeTruthy();
+    expect(device.Commands[0].Name).toBeTruthy();
+    expect(device.Commands[0].Parameters).toBeDefined();
+}
+
+var checkCommandHistory = function(device:DeviceInfo) {
+    expect(device).toBeTruthy();
+    expect(device.CommandHistory).toBeTruthy();
+    expect(device.CommandHistory.length).toBeGreaterThan(0);
+    expect(device.CommandHistory[0]).toBeTruthy();
+    expect(device.CommandHistory[0].Name).toBeTruthy();
+    expect(device.CommandHistory[0].MessageId).toBeTruthy();
+    expect(device.CommandHistory[0].CreatedTime).toBeTruthy();
+    expect(device.CommandHistory[0].Parameters).toBeTruthy();
+    expect(device.CommandHistory[0].UpdatedTime).toBeTruthy();
+    expect(device.CommandHistory[0].Result).toBeTruthy();
+    expect(device.CommandHistory[0].ErrorMessage).toBeDefined();
+}
+
+var checkTelemetry = function(device:DeviceInfo) {
+    expect(device).toBeTruthy();
+    expect(device.DeviceProperties.HubEnabledState).not.toBeNull();
+    expect(device.Telemetry).toBeTruthy();
+    expect(device.Telemetry.length).toBeGreaterThan(0);
+    expect(device.Telemetry[0].Name).toBeTruthy();
+    expect(device.Telemetry[0].DisplayName).toBeTruthy();
+    expect(device.Telemetry[0].Type).toBeTruthy();
+}
+var checkIoTHubDetailsEnabledDevice = function(device:DeviceInfo) {
+    expect(device).toBeTruthy();
+    expect(device.DeviceProperties.HubEnabledState).not.toBeNull();
+    expect(device.IoTHub).toBeTruthy();
+    expect(device.IoTHub.MessageId).toBeDefined();
+    expect(device.IoTHub.CorrelationId).toBeDefined();
+    expect(device.IoTHub.ConnectionDeviceId).toBeTruthy();
+    expect(device.IoTHub.ConnectionDeviceGenerationId).toBeTruthy();
+    expect(device.IoTHub.EnqueuedTime).toBeTruthy();
+    expect(device.IoTHub.StreamId).toBeDefined();
+}
+
 describe('devices api', () => {
     describe('get devices', () => {
+        var request = req.defaults({ json: true, baseUrl: 'https://localhost:44305/api/v1/devices' });
```

Comment:
Agree, beforeAll would make more sense since the request default instance won't need to be re-created for each spcec